domain/pricing.go

```diff
@@ -175,22 +170,27 @@ type PoolLiquidityPricerWorker interface {
 	RegisterListener(listener PoolLiquidityComputeListener)
 }
 
-// DenomPriceInfo defines the price information for the base denom.
 type DenomPriceInfo struct {
-	// Price is the price of the base denom.
-	Price osmomath.BigDec
-	// ScalingFactor is the scaling factor for the base denom.
+	Price         osmomath.BigDec
```

Comment:
I see in most of our code, we tend to use osmomath by default vs native data type

Reply:
There are 2 reasons:
- We need cross-platform determinism that native floats do not provide. As a result, we use fixed point precision decimal types
- Size of the data. Prices can be large enough to exceed the max float value

Here in SQS, since we depend on the chain APIs, we simply inherit all of the types for simplicity.